src/libraries/illink-oob.targets

```
@@ -1,63 +1,78 @@
+      <OOBLibrarySuppressionsXml Include="$(ILLinkTrimAssemblyOOBSuppressionsXmlsDir)*.xml" />
+      
       <!-- The following is the list of all the OOBs we will ignore for now -->
-      <_OOBsToIgnore Include="System.CodeDom" />
-      <_OOBsToIgnore Include="System.ComponentModel.Composition" />
-      <_OOBsToIgnore Include="System.ComponentModel.Composition.Registration" />
-      <_OOBsToIgnore Include="System.Composition.AttributedModel" />
-      <_OOBsToIgnore Include="System.Composition.Convention" />
-      <_OOBsToIgnore Include="System.Composition.Hosting" />
-      <_OOBsToIgnore Include="System.Composition.Runtime" />
-      <_OOBsToIgnore Include="System.Composition.TypedParts" />
-      <_OOBsToIgnore Include="System.Configuration.ConfigurationManager" />
-      <_OOBsToIgnore Include="System.Speech" />
-
-      <_NetCoreAppRuntimeAssemblies Include="$(NetCoreAppCurrentRuntimePath)*.dll" Exclude="$(NetCoreAppCurrentRuntimePath)*.Generator.dll;$(NetCoreAppCurrentRuntimePath)*.Native.dll;$(NetCoreAppCurrentRuntimePath)*msquic.dll" />
-      <_RuntimePackTrimmedAssemblies Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
-
-      <!-- Move previous items to FileName so that we can subtract them -->
-      <_NetCoreAppRuntimeAssembliesToFileName Include="@(_NetCoreAppRuntimeAssemblies -> '%(FileName)')">
-        <OriginalIdentity>%(Identity)</OriginalIdentity>
-      </_NetCoreAppRuntimeAssembliesToFileName>
-      <_RuntimePackAssembliesToFileName Include="@(_RuntimePackTrimmedAssemblies -> '%(FileName)')">
-        <OriginalIdentity>%(Identity)</OriginalIdentity>
-      </_RuntimePackAssembliesToFileName>
-
-      <_OOBsToTrimFileName Include="@(_NetCoreAppRuntimeAssembliesToFileName)" Exclude="@(_RuntimePackAssembliesToFileName);@(_OOBsToIgnore)" />
-      <_OOBReferencesFileName Include="@(_NetCoreAppRuntimeAssembliesToFileName)" Exclude="@(_OOBsToTrimFileName)" />
-      <_OOBsToTrim Include="@(_OOBsToTrimFileName -> '%(OriginalIdentity)')" />
-      <_OOBReferences Include="@(_OOBReferencesFileName -> '%(OriginalIdentity)')" />
-      <_OOBReferences Include="$(SystemPrivateCoreLibPath)" />
+      <OOBAssemblyToIgnore Include="System.CodeDom" />
+      <OOBAssemblyToIgnore Include="System.ComponentModel.Composition" />
+      <OOBAssemblyToIgnore Include="System.ComponentModel.Composition.Registration" />
+      <OOBAssemblyToIgnore Include="System.Composition.AttributedModel" />
+      <OOBAssemblyToIgnore Include="System.Composition.Convention" />
+      <OOBAssemblyToIgnore Include="System.Composition.Hosting" />
+      <OOBAssemblyToIgnore Include="System.Composition.Runtime" />
+      <OOBAssemblyToIgnore Include="System.Composition.TypedParts" />
+      <OOBAssemblyToIgnore Include="System.Configuration.ConfigurationManager" />
+      <OOBAssemblyToIgnore Include="System.Speech" />
+
+      <NetCoreAppCurrentAssemblyToTrim Include="$(NetCoreAppCurrentRuntimePath)*.dll"
```

I think this name is a bit misleading. Not all of the assemblies in here are "to trim", since it also contains the shared framework assemblies. In this target we are only trimming the OOB assemblies.

```suggestion
      <NetCoreAppCurrentAssembly Include="$(NetCoreAppCurrentRuntimePath)*.dll"
```

---

You are absolutely right. Fixed.